test/cli.ts

```diff
@@ -14,48 +14,100 @@
 
 import * as assert from 'assert';
 import {expect} from 'chai';
-import {factory} from '../src/factory';
+import {
+  factory,
+  ReleasePRRunResult,
+  RunResult,
+  ReleasePRMethod,
+  GitHubReleaseRunResult,
+  Method,
+  GitHubReleaseMethod,
+} from '../src/factory';
 import {GitHubRelease} from '../src/github-release';
 import {ReleasePR} from '../src/release-pr';
 import {describe, it, afterEach} from 'mocha';
 import * as sinon from 'sinon';
 
-import {parser} from '../src/bin/release-please';
+import {parser, handleError} from '../src/bin/release-please';
 import {ParseCallback} from 'yargs';
+import chalk = require('chalk');
 
 const sandbox = sinon.createSandbox();
 
+let instanceToRun: ReleasePR | GitHubRelease;
+
+function callStub(
+  instance: ReleasePR,
+  method: ReleasePRMethod
+): ReleasePRRunResult;
+function callStub(
+  instance: GitHubRelease,
+  method: GitHubReleaseMethod
+): GitHubReleaseRunResult;
+function callStub(
+  instance: ReleasePR | GitHubRelease,
+  _method: Method
+): RunResult {
+  instanceToRun = instance;
+  return Promise.resolve(undefined);
+}
+
```

Comment:
trying to be ruthless about not admitting `any` into the codebase - so I had to duplicate the overload definitions here.